### PR TITLE
Add RetryDuration in client

### PR DESCRIPTION
**Pull request by garimakhulbe**

### CODE REVIEW

File: autorest/client.go [hunk 1]
```diff
@@ -163,7 +167,7 @@ func (c Client) Do(r *http.Request) (*http.Response, error) {
 		return nil, NewErrorWithError(err, "autorest/Client", "Do", nil, "Preparing request failed")
 	}
 	resp, err := SendWithSender(c.sender(), r,
-		DoRetryForStatusCodes(c.RetryAttempts, defaultRetryInterval, statusCodesForRetry...))
```
Comment (ahmetalpbalkan) [inline autorest/client.go]: is `defaultRetryInterval` still being used?

Comment (garimakhulbe) [inline autorest/client.go]: No, we are not using it anywhere else.

Comment (garimakhulbe) [inline autorest/client.go]: I will remove this variable.

**Comment by colemickens** [comment]
Is it worth exposing this on a per-request basis? So I could get the Sender and call `Do()` or `DoWithRetryInterval(someInterval)`. For requests that I know get done faster than the RP says to retry on, but without affecting the entire client?

Might not be worth it.

This change LGTM either way.

**Pull request merged**